lib/ruby_llm/tool.rb

```diff
@@ -50,6 +50,7 @@ def parameters
 
     def name
       self.class.name
+          .gsub(/[^a-zA-Z0-9_-]/, '-')
```

Comment:
Adding `.unicode_normalize(:nfkd).encode('ASCII', replace: '')` would gracefully handle non-ASCII characters :)

Reply:
Love this method!

Reply:
Sure! Nice way to transliterate to ASCII, though would still need to handle the fact that the supported character set is very small (literally that regex), eg most symbols need converting to hyphens or underscores and accented chars not allowed.